### PR TITLE
GRO-336: Implement IP address tracking solution in Force 

### PR DESCRIPTION
**Pull request by The-Beez-Kneez**
This is to complete the work associated with the ask from Legal that required an auto-enrollment in Email Subscriptions during onboarding sign up for US users only.

We updated Metaphysics to supply us with country code based on IP address of the users during sign up, and altered the `SignUpForm` to display these conditional checkboxes.

GDPR Country Experience:
Screen Shot 2021-05-18 at 2 29 21 PM

USA Experience:
Screen Shot 2021-05-18 at 2 29 27 PM

### CODE REVIEW

File: src/typings/sharify.d.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@ declare module "sharify" {
       readonly IMAGE_LAZY_LOADING: boolean
       IS_GOOGLEBOT: boolean
       IS_MOBILE: boolean
+      IP_ADDRESS: string
```
Comment (damassi) [inline src/typings/sharify.d.ts:42]: we can also add `readonly` here

Comment (The-Beez-Kneez) [reply src/typings/sharify.d.ts]: Added.

File: src/v2/Components/Authentication/Desktop/SignUpForm.tsx [hunk 2]
```diff
@@ -164,3 +211,48 @@ export class SignUpForm extends Component<FormProps, SignUpFormState> {
...
+      variables={variables}
+      query={graphql`
+        query SignUpFormLocationQuery($ip: String!) {
+          requestLocation(ip: $ip) {
+            ...SignUpForm_requestLocation
+          }
+        }
+      `}
+      render={({ error, props }) => {
+        // what's the cooler way to do this??
```
Comment (dzucconi) [inline src/v2/Components/Authentication/Desktop/SignUpForm.tsx:240]: 😎

File: src/lib/middleware/bootstrapSharifyAndContextLocalsMiddleware.ts [hunk 3]
```diff
@@ -60,6 +60,8 @@ export function bootstrapSharifyAndContextLocalsMiddleware(
 
   updateSharifyAndContext(res, "IS_GOOGLEBOT", Boolean(ua.match(/Googlebot/i)))
 
+  updateSharify(res, "IP_ADDRESS", req.ip)
```
Comment (damassi) [inline src/lib/middleware/bootstrapSharifyAndContextLocalsMiddleware.ts:63]: Ah, it looks like @jonallured might have this commit

File: src/v2/Components/Authentication/Desktop/SignUpForm.tsx (hunk 2 above)
Comment (jonallured) [reply src/v2/Components/Authentication/Desktop/SignUpForm.tsx:240]: What are you saying the code that's there is _already_ cool?? 😝

**Pull request merged**